lightning/src/routing/router.rs

```diff
@@ -5486,6 +5529,26 @@ mod tests {
 		assert!(path_plausibility.iter().all(|x| *x));
 	}
 
+	#[test]
+	fn builds_correct_path_from_hops() {
```

Comment:
Perhaps add a scenario where there are multiple channels between some or multiple pairs of hops, and that a certain path combination gets found with those distinct channels?

Reply:
Mh, does determinism in the path actually matter? In contrary I thought we want to move towards path randomization (cf. #495) and even using alternate channels for forwarding (cf. #1278)?

Reply:
Sorry, let me clarify. The idea is not to test determinism, but rather that it is able to find multiple paths if there are multiple that match the given criteria.